In [3]:
import numpy
import pylab as plt

%matplotlib inline

In [4]:
oldbagdict = {'brown':.3, 'yellow': .2, 'red':.2, 'green':.1, 'orange':.1, 'tan':.1, 'blue':0.}
newbagdict = {'brown':.13, 'yellow': .14, 'red':.13, 'green':.2, 'orange':.16, 'tan':0., 'blue':.24}

In [72]:
print('~~~Old Bag Chances~~~'+'\n')
print('H'+'\t'+'P(H)'+'\n')
for i in range(len(oldbagdict.keys())):
    print(str(oldbagdict.keys()[i])+'\t'+str(oldbagdict.values()[i])+'\n')

~~~Old Bag Chances~~~

H	P(H)

blue	0.0

brown	0.3

yellow	0.2

green	0.1

orange	0.1

tan	0.1

red	0.2



In [44]:
print('~~~New Bag Chances~~~'+'\n')
print('H'+'\t'+'P(H)'+'\n')
for i in range(len(newbagdict.keys())):
    print(str(newbagdict.keys()[i])+'\t'+str(newbagdict.values()[i])+'\n')

~~~New Bag Chances~~~

H	P(H)

blue	0.24

brown	0.13

yellow	0.14

green	0.2

orange	0.16

tan	0.0

red	0.13



In [71]:
print('Look what we found in this undated bag!')


Look what we found in this undated bag!


In [49]:
print('Finding the Yellow Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['yellow'])+'\t'+str(.5*newbagdict['yellow']))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['yellow'])+'\t'+str(.5*oldbagdict['yellow']))

Finding the Yellow Piece

H	P(H)	P(D|H)	P(H)*P(D|H)
New Bag	.5	0.14	0.07
Old Bag	.5	0.2	0.1


In [54]:
print('(P(Yellow|Old Bag))/(P(Yellow|All)) = '+str(oldbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])))

(P(Yellow|Old Bag))/(P(Yellow|All)) = 0.588235294118


In [55]:
print('Finding the Green Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['green'])+'\t'+str(.5*newbagdict['green']))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['green'])+'\t'+str(.5*oldbagdict['green']))

Finding the Green Piece

H	P(H)	P(D|H)	P(H)*P(D|H)
New Bag	.5	0.2	0.1
Old Bag	.5	0.1	0.05


In [56]:
print('(P(Green|New Bag))/(P(Green|All)) = '+str(oldbagdict['green']/(newbagdict['green']+oldbagdict['green'])))

(P(Green|New Bag))/(P(Green|All)) = 0.333333333333


In [61]:
print('(P(Yellow|Old Bag))*(P(Green|New Bag)) = '+str(newbagdict['green']*oldbagdict['yellow']))

(P(Yellow|Old Bag))*(P(Green|New Bag)) = 0.04


In [60]:
print('(P(Green|Old Bag))*(P(Yellow|New Bag)) = '+str(oldbagdict['green']*newbagdict['yellow']))

(P(Green|Old Bag))*(P(Yellow|New Bag)) = 0.014


In [66]:
print('(P(Yellow|Old Bag))*(P(Green|New Bag))/(P(Yellow)*P(Green)) = '+str(oldbagdict['green']*newbagdict['yellow']/(oldbagdict['green']*newbagdict['yellow']+oldbagdict['yellow']*newbagdict['green'])))

(P(Yellow|Old Bag))*(P(Green|New Bag))/(P(Yellow)*P(Green)) = 0.259259259259


In [68]:
print('Finding the Yellow Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)'+'\t'+'P(H|D)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['yellow'])+'\t'+str(.5*newbagdict['yellow'])+'\t'+str(newbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['yellow'])+'\t'+str(.5*oldbagdict['yellow'])+'\t'+str(oldbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])))

Finding the Yellow Piece

H	P(H)	P(D|H)	P(H)*P(D|H)	P(H|D)
New Bag	.5	0.14	0.07	0.411764705882
Old Bag	.5	0.2	0.1	0.588235294118


In [73]:
print('Finding the Green Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)'+'\t'+'P(H|D)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['green'])+'\t'+str(.5*newbagdict['green'])+'\t'+str(newbagdict['green']/(newbagdict['green']+oldbagdict['green'])))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['green'])+'\t'+str(.5*oldbagdict['green'])+'\t'+str(oldbagdict['green']/(newbagdict['green']+oldbagdict['green'])))

Finding the Green Piece

H	P(H)	P(D|H)	P(H)*P(D|H)	P(H|D)
New Bag	.5	0.2	0.1	0.666666666667
Old Bag	.5	0.1	0.05	0.333333333333


In [85]:
newgivegreen = newbagdict['green']/(newbagdict['green']+oldbagdict['green'])
oldgivegreen = oldbagdict['green']/(newbagdict['green']+oldbagdict['green'])
newgiveyellow = newbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])
oldgiveyellow = oldbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])
print('Finding Green and Yellow Pieces in Those Bags! Where Did Ya Find Those Round Beans?!')

print('Case'+'\t'+'\t'+'Not Normalized'+'\t'+'Normalized')
print('P(G|N)*P(Y|O)'+'\t'+str(newgivegreen*oldgiveyellow)+'\t'+str(newgivegreen*oldgiveyellow/(newgivegreen*oldgiveyellow+oldgivegreen*newgiveyellow)))
print('P(Y|N)*P(G|O)'+'\t'+str(oldgivegreen*newgiveyellow))+'\t'+str(oldgivegreen*newgiveyellow/(newgivegreen*oldgiveyellow+oldgivegreen*newgiveyellow))

Finding Green and Yellow Pieces in Those Bags! Where Did Ya Find Those Round Beans?!
Case		Not Normalized	Normalized
P(G|N)*P(Y|O)	0.392156862745	0.740740740741
P(Y|N)*P(G|O)	0.137254901961	0.259259259259


The probability that they yellow M&M came from the yellow bag is .741